### ISSUE TITLE
Contributing guidelines (code style and file headers)

### ISSUE THREAD

**Issue by xen2** [xenko3d/xenko#86]
Not everybody might use resharper (and I am personally trying to use it less) so we need to have coding guidelines and file header explained somewhere.

**Comment by Kryptos-FR** [comment]
We could try to migrate most ReSharper rules to `.editorconfig`: https://docs.microsoft.com/en-us/visualstudio/ide/editorconfig-code-style-settings-reference

**Comment by xen2** [comment]
Good, I vaguely remembered there was something like this now, let's use that (and no need to write a big doc file, this could be our reference).
Ideally it would be good to enforce all code (might be some huge commits) and run checker on each commits early before too many people work on the code.

**Comment by Kryptos-FR** [comment]
There should be some CI integration possible in GitHub as well, to add checks in pull-requests.

**Comment by xen2** [comment]
@SebastianStehle would like to go this way (StyleCop analyzer and a bunch of others, enabled on CI as well). This would ensure consistent coding style and early detection of code issues.

However I was curious as to why no other big project I checked are using it (NuGet, dotnet corefx/roslyn, MonoGame, etc...).
Is it because too restrictive, too slow or something else? Do you know of any large/famous project using it in production?

**Comment by SebastianStehle** [comment]
No, unfortunately not. All companies I worked for used resharper for that. I use it for my open source project, but it is neither large nor famous: https://github.com/Squidex/squidex

XUnit has a roslyn analyzer integrated as well.

**Comment by jason-wilmans** [comment]
Yeah I use ReSharper as well. In professional and in personal development.
